spec/factories/local_authorities.rb

```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :local_authority do
-    sequence(:code) { |n| 1000 + n }
-    name { "York" }
+    sequence(:code) { |n| Tslr::SchoolEligibility::ELIGIBLE_LOCAL_AUTHORITY_CODES[n % Tslr::SchoolEligibility::ELIGIBLE_LOCAL_AUTHORITY_CODES.count] }
```

Comment:
So things get a little tricky here... Because this updated sequence is working from a fixed set of LA codes and we have a uniqueness constraint on the column, it introduces the potential for spec flakiness due to clashes. That's why the current version intentionally skips the range of known LA codes to avoid potential clashes.

One of the ways we're getting around this problem of having essentially a fixed set of identifiers for the local authorities is by using fixtures to pre-seed the test database with eligible and ineligible local authorities (and schools): https://github.com/DFE-Digital/dfe-teachers-payment-service/commit/7f3fa389ff58f7732c4136079dfe8991b0e02f35.

It can get a bit tricky when trying to mix factories and fixtures (mostly because the fixture helpers aren't available in the `FactoryBot.define` scope), but it is possible. For example, here's how the fixture for an eligible school (`:penistone_grammar_school`) could be assigned to a claim in a factory:

```ruby
FactoryBot.define do
  factory :tslr_claim_with_eligible_school, class: TslrClaim do
    claim_school { School.find(ActiveRecord::FixtureSet.identify(: penistone_grammar_school, :uuid)) }
  end
end
```

Reply:
I did wonder about that, but didn't see any good reason to have ineligible LAs by default, so thanks for that.